INSTALL.md

```diff
@@ -0,0 +1,315 @@
+# Installation
+
+## Requirements
+* [Java 1.8](http://www.oracle.com/technetwork/java/javase/downloads/index.html) or higher.
+* [Java Cryptography Extension](http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html) to
+  use Client-side Encryption with stronger ciphers (192/256-bit).
+* [Maven 3.3.x](https://maven.apache.org/) to contribute to the project.
+
+### CLI Requirements
+
+Add [BouncyCastle](http://www.bouncycastle.org/latest_releases.html) as a security provider
+ 1. Edit "$JAVA_HOME/jre/lib/security/java.security " Add an entry for BouncyCastle
+ `security.provider.11=org.bouncycastle.jce.provider.BouncyCastleProvider`
+ 2. Copy bc*.jar to $JAVA_HOME/jre/lib/ext
+
+### Unlimited Encryption Requirements
+Using stronger encryption modes (192 and 256-bit) will require installation of the
+[Java Cryptography Extensions](http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html).
+
+### Using Maven
+Add the latest java-manta-client dependency to your Maven `pom.xml`.
+
+```xml
+<dependency>
+    <groupId>com.joyent.manta</groupId>
+    <artifactId>java-manta-client</artifactId>
+    <version>LATEST</version>
+</dependency>
+```
+
+Consult [Maven Central search](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22com.joyent.manta%22%20AND%20a%3A%22java-manta-client%22)
+for a list of all available versions and dependency strings for other project types.
+
+Note: Users are expected to use the same version across sub-packages, e.g. using
+`com.joyent.manta:java-manta-client:3.0.0` with
+`com.joyent.manta:java-manta-client-kryo-serialization:3.1.0` is not supported.
+
+## From Source
+If you prefer to build from source, you'll also need
+[Maven](https://maven.apache.org/), and then invoke:
+
+``` bash
+$ mvn package
+```
+
+Which will compile the jar to `./targets/java-manta-${VERSION}.jar`. You can then
+add it as a dependency to your Java project.
+
+If you want to skip running of the test suite, use the `-DskipTests` property.
+
+# Configuration
+
+Configuration parameters can be set through either system properties or the environment. The following classes allow for
+flexible combination of parameters from different sources:
+
+  - `StandardConfigContext` leaves all values unset and provides named setters
+  - `EnvVarConfigContext` reads values from environment variables (e.g. `export MANTA_URL=value`)
+  - `SystemSettingsConfigContext` reads values from system properties (e.g. from flags `-Dmanta.url=value` or a `.properties` file)
+  - `ChainedConfigContext` allows the combination of multiple `ConfigContext`s
+
+## Parameters
+
+| Default                              | System Property                    | Environment Variable           |
```

Comment:
Why would you put the default in the middle between two settable properties?

Reply:
Good point. What about:

```
| System Property | Environment Variable | Default | 
```

I standby System Property being first -- it better fits the standards of how people read tables, with the items being defined across the top and down on the left.

Reply:
The only reason I did default first was because there is a weird truncation issue on many people's browsers when the view the table on github and they have to scroll right. I thought that the default value was the most important piece of information in that table and it shouldn't ever be hidden.

I know that is a strange thought process, but that's what it was. Honestly, I don't care that much. Let's do it at the end.

Reply:
Fine by me, it's definitely not a priority.